Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+                RunRetargetToDLL();
+                Debug.Log("Complete.");
+            }
+            catch (Exception ex)
+            {
+                Debug.LogError("Failed.");
+                Debug.LogException(ex);
+            }
+        }
+
+        private static void RunRetargetToDLL()
+        {
+            string[] allFilesUnderAssets = Directory.GetFiles(Application.dataPath, "*", SearchOption.AllDirectories);
+
+            Dictionary<string, ClassInformation> scriptFilesReferences = ProcessScripts(allFilesUnderAssets);
+            Debug.Log($"Found {scriptFilesReferences.Count} script file references.");
+
+            // DLL name to Guid
+            Dictionary<string, string> asmDefMappings = RetrieveAsmDefGuids(allFilesUnderAssets);
+
+            Dictionary<string, ClassInformation> compiledClassReferences = ProcessCompiledDLLs("PackagedAssemblies", Application.dataPath.Replace("Assets", "NuGet/Plugins/EditorPlayer"), asmDefMappings);
+            Debug.Log($"Found {compiledClassReferences.Count} compiled class references.");
+
+            Dictionary<string, Tuple<string, long>> remapDictionary = new Dictionary<string, Tuple<string, long>>();
+
+            foreach (var pair in scriptFilesReferences)
+            {
+                if (compiledClassReferences.TryGetValue(pair.Key, out ClassInformation compiledClassInfo))
+                {
+                    remapDictionary.Add(pair.Value.Guid, new Tuple<string, long>(compiledClassInfo.Guid, compiledClassInfo.FileId));
+                }
+                else
+                {
+                    // Switch to throwing exception later
+                    Debug.LogWarning($"Can't find a compiled version of the script: {pair.Key}; guid: {pair.Value.Guid}");
+                }
+            }
+
+            ProcessYAMLAssets(allFilesUnderAssets, Application.dataPath.Replace("Assets", "NuGet/Content"), remapDictionary, asmDefMappings);
+        }
+
+        private static Dictionary<string, string> RetrieveAsmDefGuids(string[] allFiles)
+        {
+            int lengthOfPrefix = Application.dataPath.IndexOf("Assets");
+            Dictionary<string, string> dllGuids = new Dictionary<string, string>();
+            foreach (string asmdefFile in allFiles.Where(t => t.EndsWith(".asmdef")))
+            {
+                string asmdefText = AssetDatabase.LoadAssetAtPath<AssemblyDefinitionAsset>(asmdefFile.Substring(lengthOfPrefix)).text;
+                string dllName = JsonUtility.FromJson<AssemblyDefinitionStub>(asmdefText).name;
+                string guid = File.ReadAllLines($"{asmdefFile}.meta")[1].Substring(6);
+                guid = CycleGuidForward(guid);
+                dllGuids.Add($"{dllName}.dll", guid);
+            }
+
+            return dllGuids;
+        }
+
+        private static void ProcessYAMLAssets(string[] allFilePaths, string outputDirectory, Dictionary<string, Tuple<string, long>> remapDictionary, Dictionary<string, string> dllGuids)
```

Comment:
>private static void ProcessYAMLAssets(string[] allFilePaths, string outputDirectory, Dictionary<string, Tuple<string, long>> remapDictionary, Dictionary<string, string> dllGuids) [](start = 8, length = 178)

Nit: It would be helpful to have a comment about the incoming dictionary types (i.e. remapDictionary is a map of what looks to be raw GUIDs i.e. non-cycled to file IDs, similar thing for the dllGuids).